services/ui-src/src/utils/forms/forms.ts

```diff
@@ -339,7 +339,8 @@ export const updateRenderFields = (
   const filteredTargetPopulations =
     removeNotApplicablePopulations(targetPopulations);
 
-  filteredTargetPopulations?.push(notApplicableOption);
+  // This one always has to be at the end for...reasons
+  filteredTargetPopulations?.push(hcbsPopulation);
```

Comment:
THOUGHT: Is this entry being added here for the first time, or added after being removed in the line above? If the latter, you might consider a `.sort` instead (guaranteed to be stable since ES2019!) Maybe something like this?

```ts
const withHcbsPopulationLast = (a: TargetPopulation, b: TargetPopulation)
  => a.id === "3Nc13O5GHA6Hc4KheO5FMSD2" ? 1 : -1;
filteredTargetPopulations.sort(withHcbsPopulationLast);
```

On second thought, that looks kinda gnarly. May not be worth doing after all. Your call!

Reply:
I am going to punt on this for now, mostly because I am afraid I don't understand the underlying reasons well enough just yet. But I _also_ feel the instinct to clean up this function a bit, so thank you for this suggestion!